src/test/unit/java/org/xpdojo/bank/TransactionTest.java

```diff
@@ -17,60 +18,62 @@
 class TransactionTest {
 
 	@Test
-	void addDepositToZeroAmount() {
-		Deposit initialDeposit = deposit(Money.ZERO);
+	void depositShouldAddDepositAmount() {
+		Deposit initialDeposit = depositOf(ZERO);
 		Transaction result = deposit(amountOf(10)).against(initialDeposit);
-		assertThat(result.getAmount(), is(amountOf(10)));
+		assertThat(result.amount(), is(amountOf(10)));
 	}
 
 	@Test
-	void addWithdrawToCauseAnAmountToBeZeroed() {
+	void withdrawalShouldSubtractWithdrawalAmount() {
 		Deposit initialDeposit = deposit(amountOf(10));
 		Transaction result = withdraw(amountOf(10)).against(initialDeposit);
-		assertThat(result.getAmount(), is(Money.ZERO));
+		assertThat(result.amount(), is(ZERO));
 	}
 	
-	@Test
-	void applyAMixOfTransactions() {
-		Transaction result = deposit(amountOf(5)).against(
-								withdraw(amountOf(10)).against(
-									deposit(amountOf(10))));
-		assertThat(result.getAmount(), is(amountOf(5)));
+	@ParameterizedTest(name = "deposit of {0} and {1} should sum to {2}")
+	@CsvSource({
+			"10, 20, 30",
+			"20, 10, 30"
+	})
+	void twoDepositsSumToTotal(long amount1, long amount2, long sum) {
+		assertThat(deposit(amountOf(amount1)).against(deposit(amountOf(amount2))).amount(), is(amountOf(sum)));
 	}
-	
+
 	@Test
-	void canSumDeposits() {
-		assertThat(deposit(amountOf(10)).against(deposit(amountOf(20))).getAmount(), is(amountOf(30)));
-		assertThat(deposit(amountOf(20)).against(deposit(amountOf(10))).getAmount(), is(amountOf(30)));
+	void withdrawalOfMoreThanInitialDepositShouldGiveNegativeTotal() {
+		assertThat(withdraw(amountOf(5)).against(deposit(amountOf(2))).amount(), is(amountOf(-3)));
 	}
 
 	@Test
-	void withdrawMoreThanTheInitialDeposit() {
-		assertThat(withdraw(amountOf(5)).against(deposit(amountOf(2))).getAmount(), is(amountOf(-3)));
+	void mixOfDepositsAndWithdrawalsShouldSumToTotalOfAllTransactions() {
+		Transaction result =
+				deposit(amountOf(5)).against(
+					withdraw(amountOf(10)).against(
+						deposit(amountOf(10))));
+		assertThat(result.amount(), is(amountOf(5)));
 	}
-	
+
 	@Test
-	void canSumMoreComplexSeriesOfDeposits() {
-		Deposit initialDeposit = deposit(amountOf(2));
-		Transaction balance =
+	void moreComplexMixOfDepositsAndWithdrawalsShouldSumToTotalOfAllTransactions() {
+		Transaction result =
 			withdraw(amountOf(5)).against(
 				deposit(amountOf(20)).against(
 					deposit(amountOf(10)).against(
-						initialDeposit
+							deposit(amountOf(2))
```

Comment:
calling this out as the initial deposit was helpful as I was all "outside in" when coding. I kept getting it muddled what's being applied to what (the method was originally called `apply`), so showing the order feels like it might be useful.

Reply:
Ah yeah, that's not a bad shout. One of the mixed transaction tests had the initial deposit stated and the other one didn't so I was making them consistent. I've got a couple more minor changes in this branch now so I'll pop the initial deposit back in both tests.